app/crud/imports.py

```
@@ -0,0 +1,154 @@
+    Import a single markdown post from content.
+    """
+    if not markdown_content.strip():
+        raise ValueError("Empty content")
+    
+    frontmatter, content = parse_markdown_frontmatter(markdown_content)
+    
+    title = frontmatter.get('title', extract_title_from_content(content))
+    if not title:
+        title = filename.replace('.md', '') if filename else "Untitled Post"
+    if not frontmatter and not content.strip():
+        raise ValueError("Content has no meaningful structure")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)